purchase_sale_inter_company/models/sale_order.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2013-Today Odoo SA
+# Copyright 2016-2019 Chafique DELLI @ Akretion
+# Copyright 2018-2019 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    auto_purchase_order_id = fields.Many2one(
+        comodel_name="purchase.order",
+        string="Source Purchase Order",
+        readonly=True,
+        copy=False,
+    )
+
+    def action_confirm(self):
+        for order in self.filtered("auto_purchase_order_id"):
+            for line in order.order_line.sudo():
+                if line.auto_purchase_line_id:
+                    line.auto_purchase_line_id.price_unit = line.price_unit
```

Comment:
I don't understand this; the user does not expect a change in the price entered in the purchase line, regardless of whether the sale price is different or not. Am I missing something?
If we want to mantain the sale price maybe we should at least log some kind of message to inform the user. There is a simililar behaviour in [account_invoice_inter_company/models/account_move.py#L125](https://github.com/OCA/multi-company/blob/16031b59e8021c32f708a79a24160aa1466b97f0/account_invoice_inter_company/models/account_move.py#L125)

Reply:
I agree with your comment, and I have a customization in place that communicates price changes between both orders and then refuses to confirm in case of a price difference, but this code has been there from the first incarnation of this module in the 8.0 branch so this is something that is best addressed in a follow-up PR and not in this migration PR.

Reply:
Here is a PR for that: https://github.com/OCA/multi-company/pull/307